bin.src/plot_astrometric_residuals.py

```diff
@@ -0,0 +1,281 @@
+    """Make a quiver plot of the astrometry error vectors."""
+    cycle = plt.rcParams['axes.prop_cycle'].by_key()['color']
+
+    color = cycle[0] if 'jointcal' in label else cycle[1]
+    scale = 0.25 if 'mean' in label else 1
+    width = 7.0
+    if focal:
+        width = .1
+        scale *= 70
+
+    fig = plt.figure(figsize=(6, 10))
+    ax = fig.add_subplot(111)
+
+    # units of the vectors are not, in fact "x", but it's simpler to treat them that way.
+    # uu, vv should be in arcsec, xx,yy in pixels or mm (if focal)
+    Q = ax.quiver(xx, yy, uu, vv, units='x', pivot='tail', scale=scale, width=width,
+                  headwidth=4, clip_on=False, color=color)
+    length = 5/scale if 'mean' in label else 100
+    key = ax.quiverkey(Q, 0.85, 0.90, length, '%s mas'%(length), angle=45,
+                       coordinates='figure', labelpos='W', fontproperties={'size': 24})
+
+    ax.set_xticklabels([])
+    ax.set_yticklabels([])
+
+    plt.title('{}'.format(ccd))
+    filename = "plots/quiver-%s-%s.png"%(ccd, label)
+    plt.savefig(filename, bbox_extra_artists=(key,))
+    plt.close(fig)
+
+
+def main():
+    parser = lsst.pipe.base.InputOnlyArgumentParser("plot_astrometric_residuals",
+                                                    description=__doc__)
+    parser.add_argument("--jointcal", action='store_true',
+                        help="Use the jointcal_wcs dataset to update each ccd's coordinates.")
+    parser.add_argument("--focalplane", action='store_true',
+                        help="Make plots and pickle files in focal plane coordinates.")
+    # We need to specify "jointcal_wcs" here so that the butler understands what a `tract` is.
+    parser.add_id_argument(name="--id", datasetType="jointcal_wcs",
+                           help="data IDs, e.g. --id visit=12345 ccd=1^2 tract=1234")
+
+    args = parser.parse_args(config=lsst.pex.config.Config())
+    name = 'jointcal' if args.jointcal else 'single'
+    butler = args.butler
+
+    if len(args.id.refList) == 0:
+        raise RuntimeError("No data to process! Check your id list and/or input data path.")
+    fluxField, newSchema, mapper, multiMatch = prep_matching(butler, args.id.refList[0])
+    goodMatches, averageCoord, distance = do_match(multiMatch, butler, args.id.refList,
+                                                   fluxField, newSchema, mapper, useJointcal=args.jointcal)
+    counts = count_ccds(goodMatches)
+
+    os.makedirs('plots', exist_ok=True)
+    os.makedirs('pickles', exist_ok=True)
+
+    for ccd in counts.keys():
+        detector = butler.get('calexp_detector', ccd=ccd)
+        xx, yy, uu, vv, bbox = compute_errors(counts, goodMatches, averageCoord, detector,
+                                              focalplane=args.focalplane)
+
+        xlim = (bbox.getMinX(), bbox.getMaxX())
+        ylim = (bbox.getMinY(), bbox.getMaxY())
+        filename = "pickles/quiverData-%s-%s.pickle"%(name, ccd)
```

Comment:
Define format template outside of for loop.  Consider making the format template a keyword-option to this function.